test/new-e2e/language-detection/language_detection_test.go

```diff
@@ -0,0 +1,111 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package languagedetection
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+)
+
+// testLocally should be set to true if running the test from a developer machine (i.e. not CI)
+// This sets the test suite to use the hardcoded agent version `versionStr`
+const testLocally = false
+
+const versionStr = "7.48.0~rc.1-1"
+
+const configStr = `
+process_config:
+  process_collection:
+    enabled: true
+  intervals:
+    process: 1
+language_detection:
+  enabled: true
+`
+
+type languageDetectionSuite struct {
+	e2e.Suite[e2e.AgentEnv]
+}
+
+func TestLanguageDetectionSuite(t *testing.T) {
+	agentParams := []func(*agentparams.Params) error{
+		agentparams.WithAgentConfig(configStr),
+	}
+
+	if testLocally {
+		agentParams = append(agentParams, agentparams.WithVersion(versionStr))
+	}
+
+	e2e.Run(t, &languageDetectionSuite{}, e2e.AgentStackDef(nil, agentParams...))
+}
+
+func (s *languageDetectionSuite) checkDetectedLanguage(command string, language string) {
+	var pid string
+	assert.Eventually(s.T(),
+		func() bool {
+			pid = s.getPidForCommand(command)
+			return len(pid) > 0
+		},
+		1*time.Second, 10*time.Millisecond,
+		fmt.Sprintf("pid not found for command %s", command),
+	)
+	require.NotEmpty(s.T(), pid)
+
+	// ensure check is able to run
+	time.Sleep(2 * time.Second)
```

Comment:
I was a bit on the fence about leaving this in or not. You're right that `assert.Eventually` will be enough to get the test to pass, but we don't need to query the agent until the process check has run and the data has been streamed to the core agent, so those cycles are "wasted".

Do you (or anyone) feel it would be better one way or the other?

Reply:
I ended up removing the sleep and relying on the `assert.Eventually` call.